modules/git/command.go

```diff
@@ -148,15 +160,15 @@ func (c *Command) Run(opts *RunOpts) error {
 		cmd.Env = opts.Env
 	}
 
-	cmd.Env = append(
-		cmd.Env,
-		fmt.Sprintf("LC_ALL=%s", DefaultLocale),
-		// avoid prompting for credentials interactively, supported since git v2.3
-		"GIT_TERMINAL_PROMPT=0",
-		// ignore replace references (https://git-scm.com/docs/git-replace)
-		"GIT_NO_REPLACE_OBJECTS=1",
-	)
+	if HomeDir == "" {
+		// TODO: now, some unit test code call the git module directly without initialization, which is incorrect.
+		// at the moment, we just use a temp HomeDir to prevent from conflicting with user's git config
+		// in future, the git module should be initialized first before use.
+		HomeDir = filepath.Join(os.TempDir(), "/gitea-temp-home")
```

Comment:
There maybe a data race here.

Reply:
It won't occur for production use. This code is only a fallback for incorrect use.

See the log (warning) below. Any one who is affected by this code block should initialize the git module first.

So the potential data race is not a problem here.